cluster/kubernetes/resource/resource.go

```diff
@@ -31,7 +34,9 @@ type baseObject struct {
 
 func (o baseObject) ResourceID() flux.ResourceID {
 	ns := o.Meta.Namespace
-	if ns == "" {
+	if o.scope == cluster.NamespaceScope && ns == "" {
+		// namespace-scoped resources with an omitted namespace
+		// implicitly map to the default namespace in the cluster
 		ns = "default"
```

Comment:
This is not necessarily the default namespace in the cluster. Yes, this was a problem before, but we now rely somewhat more on namespaces being correct.

Reply:
Oh, I didn't now. I am happy to fix it, but after some quick searches I couldn't find how to find that information (I know that `kubectl` can set a a different namespace http://www.kubetips.com/set-a-default-namespace/ , but I don't know if that's what you are referring to, if that's the case we should just provide a way to specify the namespace for flux)

Reply:
It's a setting specific to a context, given in a kubeconfig. Since the machinery for loading config is somewhat labyrinthine, I reckon the most reliable way to get it (considering we use `kubectl` for applying anyway) is to ask `kubectl`. That's what I've done in https://github.com/weaveworks/flux/pull/1442/commits/b59cef880926501029d24a68d2f603051959ae28#diff-01a1bc681d547c2ba8d68501f424b99fR371 (I'm not proud).

Reply:
is the local `kubectl` context exposed to end users?

Reply:
Yes, in the sense that you can (by design) mount one into the right place in the filesystem and it will be honoured.